kratos/geometries/geometry.h

```diff
@@ -647,11 +647,92 @@ class Geometry
     ///@name Operations
     ///@{
 
-    virtual Pointer Create( PointsArrayType const& ThisPoints ) const
+    /**
+     * @brief Creates a new geometry pointer
+     * @param ThisPoints the nodes of the new geometry
+     * @return Pointer to the new geometry
+     */
+    virtual Pointer Create(
+        PointsArrayType const& ThisPoints
+    ) const
     {
         return Pointer( new Geometry( ThisPoints, mpGeometryData ) );
     }
 
+    /**
+     * @brief Creates a new geometry pointer
+     * @param NewGeometryId the ID of the new geometry
+     * @param ThisPoints the nodes of the new geometry
+     * @return Pointer to the new geometry
+     */
+    virtual Pointer Create(
+        IndexType NewGeometryId,
+        PointsArrayType const& ThisPoints
+    ) const
+    {
+        return Pointer( new Geometry( NewGeometryId, ThisPoints, mpGeometryData));
+    }
+
+    /**
+     * @brief Creates a new geometry pointer
+     * @param NewGeometryName the name of the new geometry
+     * @param ThisPoints the nodes of the new geometry
+     * @return Pointer to the new geometry
+     */
+    virtual Pointer Create(
+        const std::string& NewGeometryName,
```

Comment:
```suggestion
        const std::string& rNewGeometryName,
```

Reply:
Ok, But I cannot change it here, or I will break the code